dependencies.yaml

```diff
@@ -343,13 +343,27 @@ dependencies:
     specific:
       - output_types: [requirements, pyproject]
         matrices:
-          - matrix: {cuda: "12.*"}
-            packages: &build_python_packages_cu12
+          - matrix:
```

Comment:
Can we simplify this using DFG's "match first" approach? This is equivalent, and far fewer lines.

```yaml
          - matrix:
              cuda: "12.*"
              cuda_suffixed: "true"
            packages:
              - rmm-cu12==24.8.*,>=0.0.0a0
          - matrix:
              cuda: "11.*"
              cuda_suffixed: "true"
            packages:
              - rmm-cu11==24.8.*,>=0.0.0a0
          - matrix:
            packages:
              - *rmm_conda
```

Reply:
Maybe we can also rename `rmm_conda` to `rmm_unsuffixed`? It's not just for conda anymore.

Reply:
Yeah totally support both of these suggestions. I'll do that here and across the other PRs.

Reply:
Ok I just pushed a commit that does both of these.

* got rid of almost all the `cuda_suffixed: "false"` blocks (one is still needed for the `pynvjitlink` vs. `cubinlink/ptxcompiler` split)
* renamed all the `*_conda` anchors to `*_unsuffixed`

I tested like this, and saw the things I expected:

```text
rapids-dependency-file-generator \
    --file-key all \
    --output requirements \
    --matrix 'cuda=11.8;cuda_suffixed=true'

rapids-dependency-file-generator \
    --file-key all \
    --output requirements \
    --matrix 'cuda=11.8;cuda_suffixed=false'

rapids-dependency-file-generator \
    --file-key all \
    --output requirements \
    --matrix 'cuda=12.2;cuda_suffixed=true'

rapids-dependency-file-generator \
    --file-key all \
    --output requirements \
    --matrix 'cuda=12.2;cuda_suffixed=false'
```